openssl3/install-sfw

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/ksh93
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2010 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+# Copyright 2021 Joyent, Inc.
+#
+
+set -o errexit
+
+LIBDIR=${DESTDIR}/lib
+BINDIR=${DESTDIR}/usr/bin
+USRLIBDIR=${DESTDIR}/usr/lib
+INCDIR=${DESTDIR}/usr/include
+
+mkdir -p $LIBDIR
+mkdir -p $BINDIR
+mkdir -p $USRLIBDIR
+mkdir -p $INCDIR
+
+if [[ -z "$VERDIR" || -z "$LIBVER" ]]; then
+	echo "VERDIR and/or LIBVER not set; fix your bugs."
+	exit 1
+fi
+
+source ../install.subr
+
+cd $VERDIR
+
+# Install libraries
+_install D libcrypto-smartos.so.${LIBVER} \
+    ${LIBDIR}/libcrypto-smartos.so.${LIBVER} 0755
+_install L libcrypto-smartos.so.${LIBVER} ${LIBDIR}/libcrypto-smartos.so
+_install L libcrypto-smartos.so.${LIBVER} ${LIBDIR}/libsunw_crypto.so
+_install D libssl-smartos.so.${LIBVER} \
+    ${LIBDIR}/libssl-smartos.so.${LIBVER} 0755
+_install L libssl-smartos.so.${LIBVER} ${LIBDIR}/libssl-smartos.so
+_install L libssl-smartos.so.${LIBVER} ${LIBDIR}/libsunw_ssl.so
+_install L ../../lib/libcrypto-smartos.so.${LIBVER} \
+    ${USRLIBDIR}/libcrypto-smartos.so.${LIBVER}
+_install L ../../lib/libcrypto-smartos.so ${USRLIBDIR}/libcrypto-smartos.so
+_install L ../../lib/libcrypto-smartos.so ${USRLIBDIR}/libsunw_crypto.so
+_install L ../../lib/libssl-smartos.so.${LIBVER} \
+    ${USRLIBDIR}/libssl-smartos.so.${LIBVER}
+_install L ../../lib/libssl-smartos.so ${USRLIBDIR}/libssl-smartos.so
+_install L ../../lib/libssl-smartos.so ${USRLIBDIR}/libsunw_ssl.so
+
+# Install commands
+## XXX KEBE ASKS, put 64-bit version in plain bin/openssl ? (for now no)
```

Comment:
This comment should also be removed, or turned into a more generalized explanation.

Reply:
There's an actual open question here that I haven't totally decided yet.  Today we deliver 32-bit openssl(1) in /usr/bin/openssl, and 64-bit openssl(1) in /usr/bin/amd64/openssl. I'm not sure if there's benefit (like there is with ls(1)) to defaulting to 32-bit, or whether or not we should just go 64-bit-only.

Reply:
I'd say the cautious route is to go 32-bit now, and we can always change it later. Eventually the whole platform will become 64-bit, but that's a problem for a different day.

Reply:
I agree, and the "code review feedback" push e430039a1f confirms the "for now no" and makes it a real decision (that lessens surprise from OpenSSL 1.0.2).  Thank you!